src/main/java/be/seeseemelk/mockbukkit/block/state/LecternMock.java

```diff
@@ -0,0 +1,62 @@
+package be.seeseemelk.mockbukkit.block.state;
+
+import be.seeseemelk.mockbukkit.inventory.InventoryMock;
+import be.seeseemelk.mockbukkit.inventory.LecternInventoryMock;
+import org.bukkit.Material;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.Lectern;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.BookMeta;
+import org.jetbrains.annotations.NotNull;
+
+public class LecternMock extends ContainerMock implements Lectern {
+
+    private int currentPage;
+
+    public LecternMock(@NotNull Material material) {
+        super(material);
+    }
+
+    protected LecternMock(@NotNull LecternMock state)
+    {
+        super(state);
+    }
+
+    @Override
+    protected InventoryMock createInventory() {
+        return new LecternInventoryMock(this);
+    }
+
+    @Override
+    public BlockState getSnapshot() {
+        return new LecternMock(this);
+    }
+
+    @Override
+    public int getPage() {
+        return this.currentPage;
+    }
+
+    @Override
+    public void setPage(int page) {
+        ItemStack book = this.getInventory().getItem(0);
+
+        int maxPages = book != null && getBookMeta(book) != null
+                ? getBookMeta(book).getPageCount()
+                : 1;
```

Comment:
This could maybe be refactored or even put into a dedicated method, but not really a necessity.

Reply:
depends on how hard to read you feel that is. Method is <10 lines which is what I aim for, but only problem with this I can see is potentially readability issues.

Reply:
The ternary operator is generally not considered the most readable asset compared to if/else structures.
Again, I said that I don't see it as a necessity, but it's generally good practice to rely on if/else structures or to make a dedicated method for this.
Just a general comment on it, it would also be fine as is.

Reply:
Be especially careful with ternary operators when using them with more complex expressions. Many people (including me) don't know of the top of their head which operations take precedence. (E.g.: is `A && B ? C : D` the same as `(A && B) ? C : D` or is it the same as `A && (B ? C : D)`. Or what about `A ? B : C + D`, is that `(A ? B : C) + D` or `A ? B : (C + D)`).

For simple situations I do use it myself though.

Reply:
I think here it should be trivial enough. Would never compound them or anything like that but I see the point.